[quant][graphmode][refactor] Returnning a cloned and modified model in FoldConvBn

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32379 [WIP][quant][graphmode] FoldConvBatchNorm2d support shared ClassTypes
* #32543 [WIP][quant][graphmode] ConvBn Folding works with bias=None
* **#32380 [quant][graphmode][refactor] Better API for fold_convbn**
* #32374 [refactor] Adding FoldConvBatchNorm2dHelper
* #32303 [quant][graphmode] Call _jit_pass_dedup_module_ueses in quantize_script
* #32302 [quant] Re-enable fold_convbn in quantize_script
* #32206 [quant] Re-enable test_nested that has different qconfig for shared ClassType
* #32205 [quant][graphmode] Support quantizing shared ClassType with different qconfigs

Summary:
We'll clone the module first and then fold conv bn and return a new
module

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19508033](https://our.internmc.facebook.com/intern/diff/D19508033)

jerryzh168 committed:
[quant][graphmode][refactor] Better API for fold_convbn

Summary:
We'll clone the module first and then fold conv bn and return a new
module

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit d18f95b9:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=d18f95b9ba6b494e8a2c41e826d905b9eb859e28).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4295571) pytorch_macos_10_13_py3_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=88003241) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=88003241))
<details>
<summary>
<code>Jan 17 16:33:29 RuntimeError: test_jit failed!</code>
</summary>

```
Jan 17 16:33:28 Ran 2341 tests in 111.034s 
Jan 17 16:33:28  
Jan 17 16:33:28 FAILED (errors=3, skipped=68, expected failures=1) 
Jan 17 16:33:28  
Jan 17 16:33:28 Generating XML reports... 
Jan 17 16:33:29 Traceback (most recent call last): 
Jan 17 16:33:29   File "test/run_test.py", line 456, in <module> 
Jan 17 16:33:29     main() 
Jan 17 16:33:29   File "test/run_test.py", line 449, in main 
Jan 17 16:33:29     raise RuntimeError(message) 
Jan 17 16:33:29 RuntimeError: test_jit failed! 
Jan 17 16:33:29 + cleanup 
Jan 17 16:33:29 + retcode=1 
Jan 17 16:33:29 + set +x 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Better API for fold_convbn"


Summary:
We'll clone the module first and then fold conv bn and return a new
module

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -1591,7 +1591,7 @@ void QuantFusion(std::shared_ptr<Graph>& graph) {
-void FoldConvBatchNorm2d(const script::Module& module) {
+script::Module FoldConvBatchNorm2d(const script::Module& module) {
   FoldConvBatchNorm2dHelper h;
   h.run(module);
```
We probably need to return the new module here now.

ZolotukhinM reviewed (approved):
Looks good modulo one comment inline. Also, please be more specific in commit messages: "better" can mean anything.

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Better API for fold_convbn"


Summary:
We'll clone the module first and then fold conv bn and return a new
module

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Better API for fold_convbn"


Summary:
We'll clone the module first and then fold conv bn and return a new
module

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Better API for fold_convbn"


Summary:
We'll clone the module first and then fold conv bn and return a new
module

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Better API for fold_convbn"


Summary:
We'll clone the module first and then fold conv bn and return a new
module

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19508033](https://our.internmc.facebook.com/intern/diff/D19508033)

[ghstack-poisoned]
